src/backend/distributed/commands/index.c

```diff
@@ -172,7 +174,21 @@ PreprocessIndexStmt(Node *node, const char *createIndexCommand)
 		return NIL;
 	}
 
-	ErrorIfUnsupportedIndexStmt(createIndexStatement);
+	if (createIndexStatement->idxname == NULL)
+	{
+		if (!CanGenerateDefaultIndexName(createIndexStatement))
+		{
+			/*
+			 * Logic that CanGenerateDefaultIndexName follows is aligned with
+			 * postgres function DefineIndex and we return NIL here to let
+			 * standart_processUtility error out if we cannot generate a default
+			 * name for index.
+			 */
+			return NIL;
+		}
+
+		createIndexStatement->idxname = GenerateDefaultIndexName(createIndexStatement);
```

Comment:
We need to be careful when modifying the CREATE INDEX statement because it may get cached across transaction when running it from PL/pgSQL. We should copy the index name into the right memory context (see what we do for schemas above) and add some PL/pgSQL tests.

Reply:
115278e48b47dd2510dc2cc52fd7483fb01e3083

Reply:
also added tests cases 0a4c8453fb4980974c012b0749ba110c26ccb0c0